specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2025-03-03/gallery.json

```diff
@@ -2966,6 +3607,248 @@
       ],
       "description": "The definition of a parameter that can be passed to a custom action of a Gallery Application Version."
     },
+    "GalleryScript": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GalleryScriptProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common-types/v1/common.json#/definitions/Resource"
+        }
+      ],
+      "description": "Specifies information about the gallery Script Definition that you want to create or update."
+    },
+    "GalleryScriptUpdate": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GalleryScriptProperties"
```

Comment:
>UpdateResourceDefinition [](http://example.com/codeflow?start=33&length=24)

This is id, name, type and tags only, which matches the common-types Resource. 

Instead of GalleryScriptUpdate, can the Patch API use GalleryScript ?

Reply:
Looks like 'location' is not required for any of the Patch operations in gallery.json. So, not changing this